### PR TITLE
Update metadata docs for ink! 4.0

### PR DESCRIPTION
**Pull request by HCastano**
There have been a couple of breaking metadata changes rolled into ink! 4.0. This PR
updates the docs to reflect that.

### CODE REVIEW

File: versioned_docs/version-4.0.0-alpha.1/basics/metadata.md [hunk 1]
```diff
@@ -303,6 +317,21 @@ not dig into them here. If you are interested in learning more take a look at th
 Other parts of the metadata, such as the `storage` object, will reference individual
 types from this type registry using the `id` key.
 
+#### `version`
+This indicates the version of the ABI format the generated metadata conforms to. This is
+district from any concept of Rust's crate versioning
```
Comment (cmichi) [inline versioned_docs/version-4.0.0-alpha.1/basics/metadata.md:322]:
```suggestion
distinct from any concept of Rust's crate versioning.
```

File: versioned_docs/version-4.0.0-alpha.1/basics/metadata.md [hunk 2]
```diff
@@ -193,6 +182,13 @@ define any events then the `events` key would contain an empty array `[]`.
 
 :::
 
+:::tip ink! 3.x Compatiblity Note
+
+The `lang_error` field was introduced as part of ink! 4.0. All ink! messages and
+constructors return a `Result` which use this as the `Error` variant.
```
Comment (cmichi) [inline versioned_docs/version-4.0.0-alpha.1/basics/metadata.md:188]:
```suggestion
constructors return a `Result` which uses this as the `Error` variant.
```

File: versioned_docs/version-4.0.0-alpha.1/basics/metadata.md [hunk 3]
```diff
@@ -185,6 +173,7 @@ The contract `spec` consists of the following **required** keys:
     - `args`: The event arguments.
     - `docs`: The event documentation.
 - `docs`: The contract documentation.
+- `lang_error`: The language specific error type.
```
Comment (cmichi) [inline versioned_docs/version-4.0.0-alpha.1/basics/metadata.md:176]:
```suggestion
- `lang_error`: The language specific error type, an error which doesn't originate from the contract itself, nor from the underlying `pallet-contracts` execution environment. See [here](https://docs.rs/ink/4.0.0-beta/ink/enum.LangError.html) for the error object that ink! returns.
```

**Review by cmichi — APPROVED** [review]
Thanks! Fine besides the above nitpicks.

File: versioned_docs/version-4.0.0-alpha.1/basics/metadata.md (hunk 3 above)
Comment (HCastano) [reply versioned_docs/version-4.0.0-alpha.1/basics/metadata.md:176]: Added to the compatibility note

**Pull request merged**